# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: xpcom/tests/gtest/TestThreadMetrics.cpp
# Commit: 593544f437fa
# Full Hash: 593544f437fa3ba3291c405aa28ab3589d21fb23
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-25 09:49:23
# Regressor Bug: 1580565
# File Overlap Count: 2
# Description:
#   Bug 1580565 - Part 6: Add a unique ID to each BrowsingContextGroup, r=kmag
#   
#   This allows us to explicitly specify BrowsingContextGroups when synchronizing
#   them. A major advantage of this is that it means we can handle an attempt to
#   create a BrowsingContext with a parent which the content process is unaware of,
# ==============================================================================

diff -r 68150e4a45af -r 593544f437fa xpcom/tests/gtest/TestThreadMetrics.cpp
--- a/xpcom/tests/gtest/TestThreadMetrics.cpp	Fri Apr 24 18:33:06 2020 +0000
+++ b/xpcom/tests/gtest/TestThreadMetrics.cpp	Fri Apr 24 18:33:09 2020 +0000
@@ -118,7 +118,8 @@
  protected:
   virtual void SetUp() {
     // building the DocGroup structure
-    RefPtr<dom::BrowsingContextGroup> group = new dom::BrowsingContextGroup();
+    RefPtr<dom::BrowsingContextGroup> group =
+        dom::BrowsingContextGroup::Create();
     mDocGroup = group->AddDocument(NS_LITERAL_CSTRING("key"), nullptr);
     mDocGroup2 = group->AddDocument(NS_LITERAL_CSTRING("key2"), nullptr);
     mCounter = mDocGroup->GetPerformanceCounter();
